src/cgame/default/cg_hud.c

```diff
@@ -1175,6 +1233,231 @@ static void Cg_DrawTargetName(const player_state_t *ps) {
+
+	GSList *list;
+
+	for (int32_t i = 0; i < notifications.num_lines; i++) {
+		list = g_slist_nth(notifications.items, i);
+
+		if (list && cgi.client->unclamped_time > ((bg_notification_item_t *) list->data)->when) {
+			notifications.items = g_slist_delete_link(notifications.items, list);
+
+			notifications.num_lines = g_slist_length(notifications.items);
+
+			i--;
+
+			continue;
+		}
+	}
+
+	if (!cg_draw_notifications->value) {
+		return;
+	}
+
+	cgi.BindFont("small", &cw, &ch);
+
+	y = (Min(notifications.num_lines - 1, cg_draw_notifications_lines->integer - 1) * NOTIFICATION_PIC_HEIGHT) + 10;
+
+	for (int32_t i = 0; i < Min(notifications.num_lines, cg_draw_notifications_lines->integer); i++) {
+		list = g_slist_nth(notifications.items, i);
+
+		if (list == NULL) {
+			continue;
+		}
+
+		item = (bg_notification_item_t *) list->data;
+
+		x = cgi.context->width - 10;
+
+		switch (item->type) {
+			case NOTIFICATION_TYPE_OBITUARY: {
+				char *name = cgi.client->client_info[item->client_id_2].name;
+
+				x -= cgi.StringWidth(name) + NOTIFICATION_PADDING_X;
+
+				cgi.DrawString(x, y + 6, name, CON_COLOR_DEFAULT);
+
+				x -= NOTIFICATION_PIC_HEIGHT + NOTIFICATION_PADDING_X;
+
+				cgi.DrawImage(x, y, ((vec_t) NOTIFICATION_PIC_HEIGHT) / HUD_PIC_HEIGHT,
+					cgi.LoadImage(Bg_GetModIcon(item->mod, item->mod & MOD_FRIENDLY_FIRE), IT_PIC));
+
+				name = cgi.client->client_info[item->client_id_1].name;
+
+				x -= cgi.StringWidth(name) + NOTIFICATION_PADDING_X;
+
+				cgi.DrawString(x, y + 6, name, CON_COLOR_DEFAULT);
+			}
+				break;
+			case NOTIFICATION_TYPE_OBITUARY_SELF: {
+				char *name = cgi.client->client_info[item->client_id_1].name;
+
+				x -= cgi.StringWidth(name) + NOTIFICATION_PADDING_X;
+
+				cgi.DrawString(x, y + 6, name, CON_COLOR_DEFAULT);
+
+				x -= NOTIFICATION_PIC_HEIGHT + NOTIFICATION_PADDING_X;
+
+				cgi.DrawImage(x, y, ((vec_t) NOTIFICATION_PIC_HEIGHT) / HUD_PIC_HEIGHT,
+					cgi.LoadImage(Bg_GetModIcon(item->mod, item->mod & MOD_FRIENDLY_FIRE), IT_PIC));
+			}
+				break;
+			case NOTIFICATION_TYPE_OBITUARY_PIC: {
+				char *name = cgi.client->client_info[item->client_id_2].name;
+
+				x -= cgi.StringWidth(name) + NOTIFICATION_PADDING_X;
+
+				cgi.DrawString(x, y + 6, name, CON_COLOR_DEFAULT);
+
+				x -= NOTIFICATION_PIC_HEIGHT + NOTIFICATION_PADDING_X;
+
+				cgi.DrawImage(x, y, ((vec_t) NOTIFICATION_PIC_HEIGHT) / HUD_PIC_HEIGHT, cgi.LoadImage(item->pic, IT_PIC));
+
+				name = cgi.client->client_info[item->client_id_1].name;
+
+				x -= cgi.StringWidth(name) + NOTIFICATION_PADDING_X;
+
+				cgi.DrawString(x, y + 6, name, CON_COLOR_DEFAULT);
+			}
+				break;
+			case NOTIFICATION_TYPE_PLAYER_ACTION: {
+				x -= cgi.StringWidth(item->string_1) + NOTIFICATION_PADDING_X;
+
+				cgi.DrawString(x, y + 6, item->string_1, CON_COLOR_DEFAULT);
+
+				x -= NOTIFICATION_PIC_HEIGHT + NOTIFICATION_PADDING_X;
+
+				cgi.DrawImage(x, y, ((vec_t) NOTIFICATION_PIC_HEIGHT) / HUD_PIC_HEIGHT, cgi.LoadImage(item->pic, IT_PIC));
```

Comment:
Precache the pic in the parse function and use that, don't keep the raw string for the pic. It's immutable, so there's no sense in having the whole string be carried along.